### PR TITLE
fix for serializers never including associations on cache hits

### PR DESCRIPTION
**Pull request by kieran**
move @node assignment and the include_associations! call outside the cached response, since including associations produces the side-effect *actually* including the associations.

Previously, subsequent renders of a cached serializer would never include associations.

### CODE REVIEW

**Comment by NullVoxPopuli** [comment]
Hi there! could you please add a test for this, so we clearly know the problem that is solved? thanks!

**Comment by chainlink** [comment]
+1

**Comment by kieran** [comment]
FYI my spec passes, but there are others which were broken as of the previous release

**Comment by gf3** [comment]
:+1:

**Comment by beauby** [comment]
I am not familiar with the 0.8 codebase so @joaomdmoura I'm assigning you (:

**Comment by jorgemanrubia** [comment]
:+1: to this, thank you so much @kieran 

Current version 0.8 is completely broken when it comes to caching objects with associations. It would be great to have this merged

File: lib/active_model/serializer.rb [hunk 1]
```diff
@@ -476,9 +479,7 @@ def scope
 
     def _serializable_hash
       return nil if @object.nil?
-      @node = attributes
```
Comment (bf4) [inline lib/active_model/serializer.rb]: Why the change of node from `attributes` to `_serializable_hash`?

Comment (kieran) [inline lib/active_model/serializer.rb]: `@node` needs to be assigned outside of `_serializable_hash`, since associations can differ based on the serializer used, and the `_serializable_hash` method itself is cached inside `serializable_hash`

Comment (bf4) [inline lib/active_model/serializer.rb]: Ok, thanks. I'm not as familiar with the 0.8 code flow.

Do you think you can get CI passing?

Comment (kieran) [inline lib/active_model/serializer.rb]: @bf4 the specs seem to fail randomly trying to call  `read_attribute_for_serialization` - the failures are not tied to versions of rails or ruby, and can change per run. The issue appears to be in the generated faker data, but I'm not exactly sure what's going on there.

Anyway, we've used this in production without issue since I opened this PR, so it's safe to merge.

Comment (bf4) [inline lib/active_model/serializer.rb]: @kieran I'm okay to merge then, since I'm trusting you, here :)  It looks like you have run the specs locally, but you say they fail intermittently for you?

Comment (kieran) [inline lib/active_model/serializer.rb]: yeah, they do about 50% of the time. Something to do with the way fake objects are generated for testing.

**Comment by NullVoxPopuli** [comment]
does this still apply? needs rebase if it does

**Comment by kieran** [comment]
rebased, should merge now
